blob/commitment_proof.go

```diff
@@ -108,6 +122,15 @@ func (commitmentProof *CommitmentProof) Verify(root []byte, subtreeRootThreshold
 		if err != nil {
 			return false, err
 		}
+
+		if len(commitmentProof.SubtreeRoots) < subtreeRootsCursor {
+			return false, fmt.Errorf("len(commitmentProof.SubtreeRoots)=%d < subtreeRootsCursor=%d",
+				len(commitmentProof.SubtreeRoots), subtreeRootsCursor)
+		}
+		if len(commitmentProof.SubtreeRoots) < subtreeRootsCursor+len(ranges) {
```

Comment:
Do we need these 2 almost identical checks?

Reply:
There aren't identical, those are 2 different conditions in there when slicing commitmentProof.SubtreeRoots  that would cause 2 separate panics hence the different checks, please see the error messages for each and please see my break down:
1. `commitmentProof.SubtreeRoots[subtreeRootsCursor:]` <-- panic 1
2. `:subtreeRootsCursor+len(ranges)]` <-- panic 2

Reply:
Cool, thanks!